core/test/matrix/bccoo.cpp

```diff
@@ -0,0 +1,802 @@
+}
+
+
+TYPED_TEST(Bccoo, CanBeEmptyBlk)
+{
+    using Mtx = typename TestFixture::Mtx;
+    auto mtx_blk = Mtx::create(this->exec, BCCOO_BLOCK_SIZE_TESTED,
+                               gko::matrix::bccoo::compression::block);
+
+    this->assert_empty_blk(mtx_blk.get());
+}
+
+TYPED_TEST(Bccoo, CanBeCreatedFromExistingDataElm)
+{
+    using value_type = typename TestFixture::value_type;
+    using index_type = typename TestFixture::index_type;
+
+    const index_type block_size = 10;
+    const index_type num_bytes = 6 + 4 * sizeof(value_type);
+    index_type ind = {};
+    gko::uint8 chunk[num_bytes] = {};
+    //    index_type offsets[] = {0, num_bytes};
+    gko::size_type offsets[] = {0, num_bytes};
+    index_type rows[] = {0};
+
+    chunk[ind++] = 0x00;
+    gko::set_value_chunk<value_type>(chunk, ind, 1.0);
+    ind += sizeof(value_type);
+    chunk[ind++] = 0x01;
+    gko::set_value_chunk<value_type>(chunk, ind, 2.0);
+    ind += sizeof(value_type);
+    chunk[ind++] = 0xFF;
+    chunk[ind++] = 0x01;
+    gko::set_value_chunk<value_type>(chunk, ind, 3.0);
+    ind += sizeof(value_type);
+    chunk[ind++] = 0xFF;
+    chunk[ind++] = 0x00;
+    gko::set_value_chunk<value_type>(chunk, ind, 4.0);
+    ind += sizeof(value_type);
+
+    auto mtx_elm = gko::matrix::Bccoo<value_type, index_type>::create(
+        this->exec, gko::dim<2>{3, 2},
+        gko::array<gko::uint8>::view(this->exec, num_bytes, chunk),
+        //        gko::array<index_type>::view(this->exec, 2, offsets),
+        gko::array<gko::size_type>::view(this->exec, 2, offsets),
+        gko::array<index_type>::view(this->exec, 1, rows), 4, block_size);
+
+    ASSERT_EQ(mtx_elm->get_num_stored_elements(), 4);
+    ASSERT_EQ(mtx_elm->get_block_size(), block_size);
+    ASSERT_EQ(mtx_elm->get_const_offsets(), offsets);
+    ASSERT_EQ(mtx_elm->get_const_rows(), rows);
+}
+
+
+TYPED_TEST(Bccoo, CanBeCreatedFromExistingDataBlk)
+{
+    using value_type = typename TestFixture::value_type;
+    using index_type = typename TestFixture::index_type;
+
+    const index_type block_size = 10;
+    const index_type num_bytes = 4 + 4 + 4 * sizeof(value_type);
+    index_type ind = {};
+    gko::uint8 chunk[num_bytes] = {};
+    //    index_type offsets[] = {0, num_bytes};
+    gko::size_type offsets[] = {0, num_bytes};
+    gko::uint8 types[] = {3};
+    index_type cols[] = {0};
+    index_type rows[] = {0};
+
+    chunk[ind++] = 0x00;
+    chunk[ind++] = 0x00;
+    chunk[ind++] = 0x01;
+    chunk[ind++] = 0x02;
+
+    chunk[ind++] = 0x00;
+    chunk[ind++] = 0x01;
+    chunk[ind++] = 0x01;
+    chunk[ind++] = 0x00;
+
+    gko::set_value_chunk<value_type>(chunk, ind, 1.0);
+    ind += sizeof(value_type);
+    gko::set_value_chunk<value_type>(chunk, ind, 2.0);
+    ind += sizeof(value_type);
+    gko::set_value_chunk<value_type>(chunk, ind, 3.0);
+    ind += sizeof(value_type);
+    gko::set_value_chunk<value_type>(chunk, ind, 4.0);
+    ind += sizeof(value_type);
```

Comment:
you can use `//` comment to describe the code and separate them.
Otherwise, you need to follow AAA rules

Reply:
Done!!